src/javaharness/java/arcs/android/client/AndroidArcsImpl.java

```diff
@@ -0,0 +1,43 @@
+package arcs.android.client;
+
+import javax.inject.Inject;
+
+import arcs.android.api.IRemotePecCallback;
+import arcs.api.Arc;
+import arcs.api.ArcsImpl;
+import arcs.api.PECInnerPort;
+import arcs.api.PECInnerPortFactory;
+
+public class AndroidArcsImpl extends ArcsImpl {
+
+  @Inject protected ArcsServiceBridge bridge;
+  @Inject protected PECInnerPortFactory pecInnerPortFactory;
+
+  @Inject
+  AndroidArcsImpl() {}
+
+  @Override
+  public void runArc(Arc arc) {
+    PECInnerPort pecInnerPort =
+        pecInnerPortFactory.createPECInnerPort(arc.getPecId(), arc.getSessionId());
+    if (arc.getParticle() != null) {
+      pecInnerPort.mapParticle(arc.getParticle());
+    }
+
+    bridge.startArc(arc, createPecCallback(pecInnerPort));
+  }
+
+  @Override
+  public void stopArc(String arcId, String pecId) {
```

Comment:
How do I get this pecId? Seems it's from Arc arc. If so, can we change this to stopArc(
Arc arc)?

Reply:
done (also AndroidArcsImpl was renamed to ArcsAndroid)